### PR TITLE
[Breaking changes] Bump several major dependencies

### PR DESCRIPTION
**Pull request by emmenko**
Follow up of #1874 

---

This PR now contains several major dependency updates, resulting in releasing a new major of appkit due to the nature of peer dependencies and the major library upgrades.

More specifically:
* React v17 is required now
* Several other peer deps cleanups, mainly to drop old versions.
* Using uikit v12
* Using webpack v5
* Using graphql v15
* Using docskit v16 (gatsby v3)

The appkit packages don't contain any breaking changes.

### CODE REVIEW

**Comment by emmenko** [comment]
FYI: the production bundle seem to work but it's the dev server that has problems.

**Comment by emmenko** [comment]
Note: re-enable website into workspaces once we are able to upgrade to Webpack v5

**Comment by emmenko** [comment]
Update: things look good now, I was able to start dev server locally and to build the production app.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "private": true,
   "scripts": {
     "prepare": "husky install",
-    "postinstall": "manypkg check && preconstruct dev && yarn compile-css-modules && yarn --cwd website && yarn --cwd website-components-playground",
+    "postinstall": "manypkg check && preconstruct dev && yarn compile-css-modules",
```
Comment (adnasa) [inline package.json:9]: 🔥

File: packages/mc-scripts/src/patches/react-dev-utils/formatWebpackMessages.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+function formatWebpackMessages(json) {
+  const formattedErrors = json.errors.map(formatMessage);
+  const formattedWarnings = json.warnings.map(formatMessage);
+  const result = { errors: formattedErrors, warnings: formattedWarnings };
+  if (result.errors.some(isLikelyASyntaxError)) {
+    // If there are any syntax errors, show just them.
+    result.errors = result.errors.filter(isLikelyASyntaxError);
+  }
+  return result;
```
Comment (adnasa) [inline packages/mc-scripts/src/patches/react-dev-utils/formatWebpackMessages.js]: this is nice!

File: packages/mc-scripts/src/build.js [hunk 3]
```diff
@@ -15,9 +15,11 @@ const fs = require('fs-extra');
 const webpack = require('webpack');
 const chalk = require('react-dev-utils/chalk');
 const checkRequiredFiles = require('react-dev-utils/checkRequiredFiles');
+// FIXME: reuse normal import after CRA supports Webpack v5.
 const formatWebpackMessages = require('react-dev-utils/formatWebpackMessages');
 const FileSizeReporter = require('react-dev-utils/FileSizeReporter');
 const printBuildError = require('react-dev-utils/printBuildError');
+// const formatWebpackMessages = require('./patches/react-dev-utils/formatWebpackMessages');
```
[the remark this answers is not in the document]
Comment (adnasa) [reply packages/mc-scripts/src/build.js]: it's redundant to L19

File: packages/mc-scripts/src/config/create-webpack-config-for-development.js [hunk 4]
```diff
@@ -397,7 +396,7 @@ module.exports = function createWebpackConfigForDevelopment(options = {}) {
 
     // For dev server
     infrastructureLogging: {
-      level: 'warn',
+      level: 'none',
     },
```
Comment (adnasa) [inline packages/mc-scripts/src/config/create-webpack-config-for-development.js]: This isn't the same as removing the option?

File: @types/commercetools-uikit__link/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  import * as H from 'history';
+  import { MessageDescriptor } from 'react-intl';
+
+  export const version: string;
+
+  // <Link>
+  export type LinkProps = {
+    tone?: 'primary' | 'inverted';
+    isExternal?: boolean;
+    to?: string | H.LocationDescriptor;
```
Comment (adnasa) [inline @types/commercetools-uikit__link/index.d.ts:12]: 🔥 yesssssssss

There is a PR for Link to TS on ui-kit..
this won't last long

File: .changeset/angry-parrots-begin.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'playground': major
+---
+
+Ships with `react@17`
```
Comment (tdeekens) [inline .changeset/angry-parrots-begin.md:5]: Now uses `react@17` and `react-dom@17`.

File: .changeset/brave-chairs-listen.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/create-mc-app': major
+---
+
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/brave-chairs-listen.md:5]:
```suggestion
- Changes required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: website-components-playground/package.json [hunk 8]
```diff
@@ -12,16 +12,16 @@
...
+    "@commercetools-uikit/design-system": "^11.2.1",
+    "@commercetools-uikit/icon-button": "^11.2.1",
+    "@commercetools-uikit/icons": "^11.2.1",
+    "@commercetools-uikit/multiline-text-field": "^11.3.0",
+    "@commercetools-uikit/secondary-button": "^11.2.1",
+    "@commercetools-uikit/select-field": "^11.2.1",
+    "@commercetools-uikit/spacings": "^11.2.1",
+    "@commercetools-uikit/text": "^11.2.1",
+    "@commercetools-uikit/text-field": "^11.2.1",
+    "@commercetools-uikit/text-input": "^11.2.1",
```
Comment (adnasa) [inline website-components-playground/package.json:24]: okay we will have conflicts.
I'm fine with merging this PR first then updating ui-kit on top of that.

File: .changeset/brown-cows-sparkle.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+---
+'@commercetools-frontend/application-shell-connectors': major
+---
+
+- Requires a peer dependency of `react@17`.
+- The `@types/react*` peer dependencies have been removed and included as normal dependencies with a minor range version.
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/brown-cows-sparkle.md:7]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: .changeset/chatty-houses-shave.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/mc-html-template': major
+---
+
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/chatty-houses-shave.md:5]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: .changeset/clever-monkeys-invite.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-local/visual-testing-app': major
+---
+
+Ships with `react@17`
```
Comment (tdeekens) [inline .changeset/clever-monkeys-invite.md:5]:
```suggestion
Ships with `react@17`Now uses react@17 and react-dom@17.
```

File: .changeset/clever-wombats-teach.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+---
+'merchant-center-application-template-starter': minor
+---
+
+- Ships template with `react@17`.
```
Comment (tdeekens) [inline .changeset/clever-wombats-teach.md:5]:
```suggestion
Now uses react@17 and react-dom@17.
```

File: .changeset/little-wasps-push.md [hunk 13]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/mc-dev-authentication': major
+---
+
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/little-wasps-push.md:5]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: .changeset/quiet-glasses-smell.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+---
+'@commercetools-frontend/application-shell': major
+---
+
+- Requires a peer dependency of `react@17`, `react-dom@17`.
+- The `@types/react*` peer dependencies have been removed and included as normal dependencies with a minor range version.
+- The peer dependency `react-intl` now only requires version `>=5`.
+- The peer dependency `@testing-library/react` now only requires version `>=11`.
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/quiet-glasses-smell.md:9]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: .changeset/sweet-jeans-draw.md [hunk 15]
```diff
@@ -0,0 +1,6 @@
+---
+'@commercetools-frontend/mc-scripts': major
+---
+
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/sweet-jeans-draw.md:5]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

File: .changeset/tricky-wolves-lick.md [hunk 16]
```diff
@@ -0,0 +1,5 @@
+---
+'@commercetools-frontend/application-config': major
+---
+
+- Change required engine version to `>=12 || >=14`.
```
Comment (tdeekens) [inline .changeset/tricky-wolves-lick.md:5]:
```suggestion
- Change required Node.js engine version to `>=12 || >=14` in `package.json`.
```

**Review by tdeekens — APPROVED** [review]

**Review by adnasa — APPROVED** [review]
Let's spin this locally, one time.

Regarding the UI-kit upgrade, we'll run into conflicts.
either the ui-kit upgrade goes first or this PR. I'm fine with either, but perhaps it's easier that this PR goes first.

**Review by adnasa — APPROVED** [review]
🚀

**Pull request merged**